### ISSUE TITLE
Fatal error on block 12785713

### ISSUE THREAD

**Issue by pioul** [curvefi/curve-subgraph#4]
Opening this issue to let indexers know we're currently aware and investigating the following error, which causes indexing to fail at block 12785713:

```
{
  "data": {
    "indexingStatusForCurrentVersion": {
      "fatalError": {
        "block": {
          "number": "12785713"
        },
        "handler": null,
        "message": "Heap access out of bounds. Offset: 42 Size: 6291556\twasm backtrace:\t    0: 0x1c74 - <unknown>!generated/schema/Gauge#save\t    1: 0x1dd6 - <unknown>!src/mappings/dao/gauge-controller/handleNewGauge\t in handler `handleNewGauge` at block #12785713 (d64b643a6699097df4b95a36282a9ec776bd89619b32194d44401d5a4e621464)"
      },
      "nonFatalErrors": [],
      "subgraph": "Qma85KVRwTpstXgcHhmufP8Psc54AYKdYfZSrotx89j3jN"
    }
  }
}
```

We'll ensure a proper fix is found and deployed asap. If you have more info from your own logs don't hesitate to share them below!

**Comment by mindstyle85** [comment]
mine actually stops here:

```
{
  "data": {
    "indexingStatuses": [
      {
        "chains": [
          {
            "chainHeadBlock": {
              "number": "12818953"
            },
            "lastHealthyBlock": null,
            "latestBlock": {
              "number": "10690156"
            },
            "network": "mainnet"
          }
        ],
        "fatalError": {
          "message": "`ipfs.cat` is deprecated. Improved support for IPFS will be added in the future\twasm backtrace:\t    0: 0x1eab - <unknown>!src/mappings/dao/voting/handleStartVote\t in handler `handleStartVote` at block #10690157 (e6bbfad9dd8df1eb44fb61cee8163efc618ac05e0cc7a4a626b4df74e85d90ce)"
        },
        "health": "failed"
      }
    ]
  }
}
```

also a fatal error.

**Comment by sistemico** [comment]
Testing a few patches to solve all those issues. As long as test passes (waiting for full indexing on dev environment) we will make a new deployment.

**Comment by pioul** [comment]
Just published v0.2.0 that fixes and improves a bunch of things thanks to work from @sistemico!
